xonsh/prompt/vc_branch.py

```diff
@@ -125,9 +125,9 @@ def current_branch(pad=NotImplemented):
                       "and will be removed in the future")
     branch = None
     cmds = builtins.__xonsh_commands_cache__
-    if cmds.lazy_locate_binary('git'):
+    if cmds.lazy_locate_binary('git') or cmds.locate_binary('git'):
         branch = get_git_branch()
-    if cmds.lazy_locate_binary('hg') and not branch:
+    if cmds.lazy_locate_binary('hg') or cmds.locate_binary('hg') and not branch:
```

Comment:
If I haven't got mercurial installed, then this will search my entire path for the `hg` binary for every prompt. 

I think the only solution is to have the `cmd_cache` update on xonsh startup in a separate thread. Then when the command cache is populated the branch labels will start to work. Of course the branch may then not work on the first prompt, but that the price to pay for fast startup.